jerry-core/ecma/builtin-objects/ecma-builtin-array-iterator-prototype.c

```diff
@@ -92,11 +92,26 @@ ecma_builtin_array_iterator_prototype_object_next (ecma_value_t this_val) /**< t
   else
   {
 #endif /* !CONFIG_DISABLE_ES2015_TYPEDARRAY_BUILTIN */
-    JERRY_ASSERT (ecma_get_object_type (array_object_p) == ECMA_OBJECT_TYPE_ARRAY);
```

Comment:
So the problem is that we get a regular object here instead of an array, right? What if we change the assert into an if, retrieve the length from the extended object if we have an array, and if not, then fall back to the code you just added?

Reply:
Yes it is. The `ecma_op_object_get` contains a fast path for the array length property which does exactly the same as you suggested so I think it is not a big overhead to use this function here.